pkg/generators/servers.go

```diff
@@ -809,3 +843,21 @@ func (g *ServersGenerator) mapMethodNameToHTTPMethod(name *names.Name) string {
 		return http.MethodGet
 	}
 }
+
+func (g *ServersGenerator) readerName(name *names.Name) string {
```

Comment:
This function should receive as parameter the `Parameter`  or the `Type`, not the name. And it should compare the type with the type objects from the model, not just the names. See the `ZeroValue` method in the `pkg/golang/types.go` file for an example.

Reply:
Done.